src/adagrams.js

```diff
@@ -1,15 +1,129 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams {
+  static LETTERS = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  static SCORE_CHART = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10,
+  };
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  static drawLetters() {
+    let chars = [];
+    for (let char in Adagrams.LETTERS) {
+      let repeat = Adagrams.LETTERS[char];
+      while (repeat > 0) {
+        chars.push(char);
+        repeat -= 1;
+      }
+    }
+    let hand = [];
+    while (hand.length < 10) {
+      let rand_letter = chars[Math.floor(Math.random() * chars.length)];
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+      if (chars.includes(rand_letter)) {
+        hand.push(rand_letter);
+        chars.splice(chars.indexOf(rand_letter), 1);
+      }
```

Comment:
This works, but there's some extra work here.  This is finding a letter and then removing the first instance of the letter from the array, which requires an extra lookup to find the first index.  One optimization would be to store the result of the Math.floor(...) call in a variable and splice out that index.